### PR TITLE
Build and test with Python 3.12

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,7 +34,9 @@ skip = [
   "*-manylinux_i686",
   "pp*", 
   "*-musllinux_*",
-  "cp312*"
+  "cp36*"
+  "cp37*"
+  "cp38*"
```
Comment (jtilly) [inline pyproject.toml:39]:
```suggestion
  "cp36*",
  "cp37*",
  "cp38*"
```

**Comment by xhochy** [comment]
Depends on https://github.com/conda-forge/numexpr-feedstock/pull/55

**Comment by MarcAntoineSchmidtQC** [comment]
@xhochy, I fixed one of the failing checks (windows wheel), but the osx64 build failure is still happening. Numexpr has been updated, so I think the problem comes from somewhere else.

**Comment by MarcAntoineSchmidtQC** [comment]
I fixed the last failing job. We needed to update the macos clang/llvm version to 16.

**Review by MarcAntoineSchmidtQC — APPROVED** [review]

**Comment by MarcAntoineSchmidtQC** [comment]
I approved but would like someone to check the last 2 commits.

**Review by jtilly — APPROVED** [review]

**Pull request merged**